### DISCUSSION TITLE
ZBBridge: IKEA buttons (E1743, E1812) do not report battery percentage

### DISCUSSION THREAD

**Discussion by moodyblue** [arendst/Tasmota#13470] [Q&A] [answered]
DO NOT DELETE ANY TEXT from this template! Otherwise the issue will be auto-closed.
-->

### PROBLEM DESCRIPTION
I have some ikea and xiaomi devices paired with a Sonoff ZBBridge. All but Ikea buttons E1743/E1812 show battery icon in Tasmota main screen and report battery percentage in MQTT.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Sonoff ZBBridge paired with Ikea buttons E1743 and E1812
- [x] Tasmota binary firmware version number used: 10.0.0 + ncp-uart-nsw_6.7.9_115200.ota 
  - [x] Pre-compiled
  - [ ] Self-compiled
- [x] Flashing tools used: Web
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [x] Provide the output of this command: `Status 0`:
```lua
20:54:24.217 MQT: stat/zbb_gf/STATUS = {"Status":{"Module":0,"DeviceName":"Office","FriendlyName":["ZBB_Office1"],"Topic":"zbb_gf","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}
20:54:24.223 MQT: stat/zbb_gf/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota-minimal.bin","RestartReason":"Power On","Uptime":"0T08:55:41","StartupUTC":"2021-10-25T10:58:43","Sleep":50,"CfgHolder":4617,"BootCount":120,"BCResetTime":"2021-02-04T20:41:14","SaveCount":338,"SaveAddress":"F9000"}}
20:54:24.228 MQT: stat/zbb_gf/STATUS2 = {"StatusFWR":{"Version":"10.0.0(zbbridge)","BuildDateTime":"2021-10-19T08:23:51","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":160,"Hardware":"ESP8266EX","CR":"403/699"}}
20:54:24.232 MQT: stat/zbb_gf/STATUS3 = {"StatusLOG":{"SerialLog":0,"WebLog":2,"MqttLog":3,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["Aveiro-Domotic",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C8000100060000005A00000000000000","00000080","40006082","00004000"]}}
20:54:24.241 MQT: stat/zbb_gf/STATUS4 = {"StatusMEM":{"ProgramSize":670,"Free":1120,"Heap":25,"ProgramFlashSize":2048,"FlashSize":2048,"FlashChipId":"1540A1","FlashFrequency":40,"FlashMode":3,"Features":["00000809","0F1007C6","04400001","00000003","00000000","00000000","00020080","00200000","00000000"],"Drivers":"1,2,4,7,9,10,12,20,23,38,41,50","Sensors":"1"}}
20:54:24.246 MQT: stat/zbb_gf/STATUS5 = {"StatusNET":{"Hostname":"zbb-gf-0827","IPAddress":"192.168.129.132","Gateway":"192.168.129.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.129.1","DNSServer2":"0.0.0.0","Mac":"A4:CF:12:DC:03:3B","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
20:54:24.250 MQT: stat/zbb_gf/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.130.1","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_DC033B","MqttUser":"Melga","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
20:54:24.256 MQT: stat/zbb_gf/STATUS7 = {"StatusTIM":{"UTC":"2021-10-25T19:54:24","Local":"2021-10-25T20:54:24","StartDST":"2021-03-28T01:00:00","EndDST":"2021-10-31T01:00:00","Timezone":99,"Sunrise":"07:26","Sunset":"17:41"}}
20:54:24.261 MQT: stat/zbb_gf/STATUS10 = {"StatusSNS":{"Time":"2021-10-25T20:54:24"}}
20:54:24.267 MQT: stat/zbb_gf/STATUS11 = {"StatusSTS":{"Time":"2021-10-25T20:54:24","Uptime":"0T08:55:41","UptimeSec":32141,"Vcc":3.438,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"Aveiro-Domotic","BSSId":"7A:83:C2:9A:8B:04","Channel":11,"Mode":"11n","RSSI":60,"Signal":-70,"LinkCount":1,"Downtime":"0T00:00:03"}}}
```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### SCREENSHOTS
All devices have battery icons and report BatteryPercentage in MQTT, except `button-garage` (a ikea button E1812)
![image](https://user-images.githubusercontent.com/6073615/138763385-ba224895-78ba-4d75-922a-8af062272c6e.png)

### ADDITIONAL CONTEXT
Zigbee2MQTT documentation says that these devices report battery status:
https://www.zigbee2mqtt.io/devices/E1743.html
https://www.zigbee2mqtt.io/devices/E1812.html

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by sfromis** [comment]
Yes, not all Zigbee devices will use battery on reporting battery state all the time. Not a bug. Using a command like this, you may be able to query the battery state:
`ZbSend {"Device":"Ikea-Button","Read":{"BatteryPercentage":true}}`

**Comment by moodyblue** [comment]
Thanks. How can I parse it from this output
```
21:42:27.497 CMD: ZbSend {"Device":"button-garage","Read":{"BatteryPercentage":true}}
21:42:27.504 MQT: stat/zbb_an/RESULT = {"ZbSend":"Done"}
21:42:35.210 ZIG: {"ZbEZSPReceived":"80004226AB"}
21:42:35.217 MQT: tele/zbb_an/RESULT = {"ZbRouteError":{"ShortAddr":"0xAB26","Status":66,"StatusMessage":"MAC_INDIRECT_TIMEOUT"}}
21:42:44.438 ZIG: {"ZbEZSPReceived":"80004226AB"}
21:42:44.443 MQT: tele/zbb_an/RESULT = {"ZbRouteError":{"ShortAddr":"0xAB26","Status":66,"StatusMessage":"MAC_INDIRECT_TIMEOUT"}}
21:42:53.667 ZIG: {"ZbEZSPReceived":"80004226AB"}
21:42:53.674 MQT: tele/zbb_an/RESULT = {"ZbRouteError":{"ShortAddr":"0xAB26","Status":66,"StatusMessage":"MAC_INDIRECT_TIMEOUT"}}
```
But zigbee2mqtt claims to have it available. Why not Z2T ?

**Comment by ascillato** [comment]
ZbRouteError means that your device has a communication error with the ZigBee Bridge and that your command didn't reach the device. Please, redo the pairing and check if the battery is Ok. Your device last report was 5 hours ago, so it may not have battery left or it is disconnected.

Battery Status is not being sent by ZigBee Devices all the time. Sometimes they send once a day.

**Comment by sfromis** [comment]
Zigbee2Tasmota simply relays what the device sends, each time, unlike Zigbee2MQTT which may combine things.

**Comment by moodyblue** [comment]
> ZbRouteError means that your device has a communication error with the ZigBee Bridge and that your command didn't reach the device.

I'm puzzled. The device answers `ZbMap`, so it is correctly paired.

**Reply by ascillato** [reply]
ZbMap is a command for the Coordinator, not for the device. The error MAC_INDIRECT_TIMEOUT means that the coordinator is not reaching your device. Please, do the test of pairing it again and send the command that @sfromis have told you.

**Comment by moodyblue** [comment]
> 
> 
> Zigbee2Tasmota simply relays what the device sends, each time, unlike Zigbee2MQTT which may combine things.

It would be nice to have consistent behaviour. Is this possible ? I've not found a list of Z2T supported devices so I'm relying (probably I'm wrong) in the zigbee2mqtt list.

**Reply by sfromis** [reply]
The list is here: https://zigbee.blakadder.com/

Note that absence of the Tasmota logo for a device does not mean "not supported", merely not confirmed working. In general, I'd expect devices to work, but there might be a few exceptions, and you may have to explore what commands are supported.

**Reply by moodyblue** [reply]
Thanks. E1743 and E1812 appear on that list (as remotes), and both are supposed to support battery percentage. I'm using OpenHab and it would be easy to develop a rule to query battery once a day. The question is: what command shall I send ? `ZbSend {"Device":"Ikea-Button","Read":{"BatteryPercentage":true}}` does not work.

**Reply by sfromis** [reply]
That command has worked on an Ikea E1743 on/off button of mine, giving:
`{"Ikea-xxxx":{"Device":"0xXXXX","Name":"Ikea-xxxx","BatteryPercentage":11,"Endpoint":1,"LinkQuality":123}}`

**Reply by ascillato** [reply]
The error MAC_INDIRECT_TIMEOUT means that the coordinator is not reaching your device. Please, do the test of pairing it again and send the command that @sfromis have told you.

**Reply by moodyblue** [reply]
Looks like the device is disconnected most of the time. I've repeated the `ZbSend` command several times and one was successfull.

I have not yet paired the device again, but this will be my next test.

**Comment by moodyblue** [comment, marked as answer]
I've paired both buttons over and over again. They only answer the `ZbSend {"Device":"Ikea-Device","Read":{"BatteryPercentage":true}}` command if, and only if, I press the button during the 27second search cycle. Otherwise I get 3x `MAC_INDIRECT_TIMEOUT`

**Reply by sfromis** [reply]
Well, while the device chooses to be offline, to reduce battery drain, you cannot reach it.

**Reply by moodyblue** [reply]
> 
> 
> Well, while the device chooses to be offline, to reduce battery drain, you cannot reach it.

I understand that. But you've said previously that the command works for you. Are you using the same tasmota and Z2T version ? If I don't have a reliable method to inquire the battery status perhaps a tasmota rule, triggered when the device decides to send something,  is my only hope of doing it.

**Reply by sfromis** [reply]
Well, the command works excellently, as long as the device is online. If I wanted to use extra battery power on reporting battery status, I'd indeed make a small automation firing off a query after the button was operated, unless "too short" time since last query.

Strictly speaking I did not say that it "works for me", but carefully selected my words to reflect my reality, that it "has worked". The difference is that I'm not doing it on a regular basis, but only manually out of rare curiosity.

**Reply by moodyblue** [reply]
Thanks. All my Xiaomi/Aqara buttons and sensors report battery status periodically. If I knew that Ikea buttons did not do it with Z2T I would probably never bought them. 

But it's not the end of the world, probably I will be able to implement a method to collect battery status periodically from Ikea buttons.

I've started this thread as a possible Z2T bug because zigbee2mqtt reports battery status periodically and I thought that Z2T would do the same. But if this is not accepted neither as a bug nor as an improvement suggestion no problem.

**Reply by sfromis** [reply]
The case may be that Z2M does extra work to fire off commands to query battery state, while consuming extra battery power to do this. The absence of such extra function cannot in any way be classified as a bug.

**Reply by s-hadinger** [reply]
Indeed ikea never spontaneously report battery. I tried to make them report regularly but it was a disaster and drained the battery over a few days. 

If you really want it, you could write a rule so that you query battery status each time a button is pressed on the device. But then don't complain about battery drain.

**Reply by sfromis** [reply]
@s-hadinger What can get confusing is that the device status on the Z2T Tasmota front page still reports battery numbers seen a long time ago, even across restarts, leading to the discrepancy that what's presented as "current status" appears flawed, badly out of sync with reality.

**Reply by s-hadinger** [reply]
I see. Thanks. Unfortunately I don't store the timestamp of the battery reading. Something to add.

**Reply by moodyblue** [reply]
I don't need to read the battery continuously. One week interval is perfectly acceptable. Or use an infrequent action as trigger. E1743 and E1812 report `DimmerMove` and `DimmerStop`. IMO `DimmerMove` can be used as a trigger to request battery status.

**Reply by sfromis** [reply]
For weekly checks, you could also have a timer to set a flag, and a rule to query battery state upon next action from the button, when the flag is set. Easy to manage if you use the reception of battery state to reset the flag.

**Reply by MattWestb** [reply]
Many user with Sonoff ZBB (and some deCONZ) was / is having extreme battery draining in tasmota and ZHA (48  hours -> Zero = looping firmware update) with IKEA controllers and also Philips HUE remotes and we have not finding one 110% working way for fixing that.

One important thing is using the latest firmware in the Zigbee module (.EZSP 6.7.9.0) then it fixing the RF tuning that is not calibrated in the chip (shall have bin made in the factory).

Also having some good routers around so the end device can having them as parent looks being the best way getting it working (I have blocking my "IKEA Billy EZSP" having direct children = all end device must using router as parent and it working great).

I have not having time testing if the newly released IKEA firmware is helping or not then i have many devices locked in the bad test firmware that need being re-flashed with SWD for getting the last production version installed but i can being that can helping 2.

Also IKEA controller is reporting the battery OK then setting up attribute reporting but the value reported is not being updated if not pressing on button = if you have one switch in the basement storage and you is using it 1 time in the mouths its reporting the same battery for 1 mouths and is updating it then you is going down and using it and then reporting the new for one mouths = device personality.

**Reply by moodyblue** [reply]
Thanks. I've suffered from the battery drain with EZSP 6.7.6 but 6.7.8 solved it. I've moved to EZSP 6.7.9 one week ago and seems to be ok.

I have two Ikea buttons, one is near the ZBB (4 meters, open distance, no obstacles) and the other near an Ikea lamp (4 meters, open distance, no obstacles). ZB map reports the last one as connected through the lamp. I have no idea of what is their firmware and I don't know how to update it.

I was planning to install other Ikea buttons throughout the house but I will probably return them to Ikea and use Xiaomi Mijia WXKG01LM instead. Their price is 50% higher but affordable. I have one of them (in fact, it was my first ZB button ever) and it works very very well.

**Reply by MattWestb** [reply]
Great !!

Its one problem with the WXKG01LM its no one light controller so its not sending commands to bounded light groups if you like steering light that also working the the system is offline. If you dont need bounding its OK and using Xiaomi and using "clicks" for automatons that only woks then the system is 100% working online.

If you like you can setting ZHA profile in tasmota and install ZHA and doing the upgrade if your IKEA device and then changing back to ZBB profile (i have not testing if all is safe and not loosing the devices between the switching but shall working in the theory, so no grantee).

**Reply by moodyblue** [reply]
All my automations use OpenHab. Some buttons control several lights but that's managed by OH. And if OH fails I can revert to the original switches / remotes. I have one (and only one so far) tasmota rule to manage a Sonoff 4ch (irrigation system) in case OH fails.

I've tried ZHA some months ago but it forced me to pair devices again. Unless there is a real benefit by upgrading Ikea's firmware I'm not planning to do it in the near future. If you are planning to make such upgrade pls share your experience.

**Comment by xjvi** [comment]
You can force the IKEA E1743 to give the battery percentage by sending the command below directly after receiving a message from e.g. a button press.
ZbSend {"Device":"Ikea-Device","Read":{"BatteryPercentage":true}}

Console log after pressing the "I" button on the IKEA E1743:
18:43:54.996 MQT: tele/ST-ZBridge1/TDK2/SENSOR = {"ZbReceived":{"TDK2":{"Device":"0xC0F9","Name":"TDK2","0006!01":"","Power":1,"Endpoint":1,"LinkQuality":71}}}
18:43:55.711 CMD: ZbSend {"Device":"TDK2","Read":{"BatteryPercentage":true}}
18:43:55.717 MQT: stat/ST-ZBridge1/RESULT = {"ZbSend":"Done"}
18:43:57.431 MQT: tele/ST-ZBridge1/TDK2/SENSOR = {"ZbReceived":{"TDK2":{"Device":"0xC0F9","Name":"TDK2","BatteryPercentage":87,"Endpoint":1,"LinkQuality":76}}}
